sktime/datatypes/_utilities.py

```diff
@@ -392,7 +392,15 @@ def update_data(X, X_new=None):
             return np.concatenate([X, X_new], axis=2)
     #  if y is pandas, we use combine_first to update
     elif isinstance(X_new, (pd.Series, pd.DataFrame)) and len(X_new) > 0:
-        return X_new.combine_first(X)
+        # using .combine_first() can mess with df frequencies, e.g. change 24H to 1D,
+        # which can cause issues when converting fh.to_absolute() at the predict stage
+        # because to_absolute(self.cutoff) converts timestamps to period internally
+        X_new = X_new.combine_first(X)
+        if hasattr(X.index, "freq") and hasattr(X_new.index, "freq"):
```

Comment:
hm, I assume you have a specific situation in mind here - could you add a test for it?
This feels like one of the things that `pandas` might plausibly break in minor updates, so it would be good to have some explicit coverage.

Reply:
added as above